Source/WebInspectorUI/UserInterface/Views/ElementsTabContentView.js

```diff
@@ -127,8 +132,77 @@ WI.ElementsTabContentView = class ElementsTabContentView extends WI.ContentBrows
         super.initialLayout();
 
         this.element.appendChild(WI.ReferencePage.ElementsTab.DOMTree.createLinkElement());
+        if(WI.settings.experimentalEnableAccessibilityTreeView.value)
+            this.element.appendChild(this.createAccessibilityTreeButton());
+    }
+
+    createAccessibilityTreeButton()
+    {
+        let wrapper = document.createElement("span");
+        wrapper.className = "accessibility-tree-container";
+
+        let treeToggleButton = wrapper.appendChild(document.createElement("button"));
+        treeToggleButton.type = "button";
+        treeToggleButton.className = "accessibility-tree-button";
+
+        let accessibilityTreeButtonIcon = document.createElement("img");
+        accessibilityTreeButtonIcon.src = "Images/Accessibility.svg";
+        accessibilityTreeButtonIcon.alt = "Accessibility Tree";
+        accessibilityTreeButtonIcon.className = "accessibility-tree-icon";
+        treeToggleButton.appendChild(accessibilityTreeButtonIcon);
+
+        const showAccessibilityTooltip = WI.UIString("Go to Accessibility Tree");
+        const showDomTooltip = WI.UIString("Go to DOM Tree");
+        treeToggleButton.title = showAccessibilityTooltip;
+        treeToggleButton.addEventListener("click", (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+            this._toggleTreeView(accessibilityTreeButtonIcon, treeToggleButton, showAccessibilityTooltip, showDomTooltip);
+        });
+
+        return wrapper;
     }
 
+    _toggleTreeView(accessibilityTreeButtonIcon, treeToggleButton, showAccessibilityTooltip, showDomTooltip)
+    {
+        let mainDOMTree = WI.networkManager.mainFrame?.domTree;
+        if (!mainDOMTree)
+            return;
+        
+        let contentBrowser = this.contentBrowser;
+        let currentView = contentBrowser.currentContentView;
+        if (currentView?.domTreeOutline)
+            this._lastSelectedNode = currentView.domTreeOutline.selectedDOMNode();
+
+        WI.ElementsTabContentView._accessibilityTreeOn = !WI.ElementsTabContentView._accessibilityTreeOn;
```

Comment:
Holding state in a view like this isn't great. Model objects are better (though I'm not sure which one is best yet). The underscore implies a "private" member, but this is accessed in DOMTreeElement.

This might become obsolete if you adjust how the view gets shown.